xla/service/gpu/tests/gpu_fused_mha_test.cc

```diff
@@ -1813,6 +1813,114 @@ XLA_TEST_F(FlashAttentionBMMScaleSoftmaxDropoutBMM,
+      call.81 = bf16[1,1,128,128]{3,2,1,0} call(divide.82, constant.5, constant.4), to_apply=clip.33
+      convert.82 = f8e4m3fn[1,1,128,128]{3,2,1,0} convert(call.81)
+      convert.83 = bf16[1,1,128,128]{3,2,1,0} convert(convert.82)
+      score.0 = f32[1,1,128]{2,1,0} parameter(5)
+      custom-call.7 = (bf16[1,1,128,128]{3,2,1,0}, bf16[1,1,128,128]{3,2,1,0}, bf16[1,1,128,128]{3,2,1,0}, u8[0]{0}) custom-call(convert.19, convert.31, convert.43, score.0, convert.83, /*index=5*/convert.73), custom_call_target="__cudnn$fmhaSoftmaxBackward", operand_layout_constraints={bf16[1,1,128,128]{3,2,1,0}, bf16[1,1,128,128]{3,2,1,0}, bf16[1,1,128,128]{3,2,1,0}, f32[1,1,128]{2,1,0}, bf16[1,1,128,128]{3,2,1,0}, bf16[1,1,128,128]{3,2,1,0}}, api_version=API_VERSION_STATUS_RETURNING, backend_config={"operation_queue_id": "0", "wait_on_operation_queues": [], "cudnn_fmha_backend_config": {"algorithm": {"algo_id": "0", "math_type": "TENSOR_OP_MATH", "tuning_knobs": {"17": "1", "24": "0"}, "is_cudnn_frontend": true, "workspace_size": "0"}, "fmha_scale": 1.0, "dropout_rate": 0.0, "intermediate_tensor_shape": {"element_type": "BF16", "dimensions": ["1", "1", "128", "128"], "tuple_shapes": [], "layout": {"dim_level_types": [], "dim_unique": [], "dim_ordered": [], "minor_to_major": ["3", "2", "1", "0"], "tiles": [], "element_size_in_bits": "0", "memory_space": "0", "index_primitive_type": "PRIMITIVE_TYPE_INVALID", "pointer_primitive_type": "PRIMITIVE_TYPE_INVALID", "dynamic_shape_metadata_prefix_bytes": "0"}, "is_dynamic_dimension": [false, false, false, false]}, "seed": 42, "is_flash_attention": true, "mask_type": "NO_MASK", "sliding_window_length": 0, "bmm1_grad_gemm1_dot_dimension_numbers": {"lhs_contracting_dimensions": ["2"], "rhs_contracting_dimensions": ["2"], "lhs_batch_dimensions": ["0", "1"], "rhs_batch_dimensions": ["0", "1"]}, "bmm1_grad_gemm2_dot_dimension_numbers": {"lhs_contracting_dimensions": ["3"], "rhs_contracting_dimensions": ["2"], "lhs_batch_dimensions": ["0", "1"], "rhs_batch_dimensions": ["0", "1"]}, "bmm2_grad_gemm1_dot_dimension_numbers": {"lhs_contracting_dimensions": ["2"], "rhs_contracting_dimensions": ["2"], "lhs_batch_dimensions": ["0", "1"], "rhs_batch_dimensions": ["0", "1"]}, "bmm2_grad_gemm2_dot_dimension_numbers": {"lhs_contracting_dimensions": ["3"], "rhs_contracting_dimensions": ["3"], "lhs_batch_dimensions": ["0", "1"], "rhs_batch_dimensions": ["0", "1"]}}}
+      ROOT get-tuple-element.8 = bf16[1,1,128,128]{3,2,1,0} get-tuple-element(custom-call.7), index=0
+    })";
+
+  std::string hlo_string = R"(
+HloModule jit__unnamed_wrapped_function_, entry_computation_layout={(bf16[1,1,128,128]{3,2,1,0}, bf16[1,1,128,128]{3,2,1,0}, bf16[1,1,128,128]{3,2,1,0}, bf16[1,1,128,128]{3,2,1,0}, bf16[1,1,128,128]{3,2,1,0}, f32[1,1,128]{2,1,0})->bf16[1,1,128,128]{3,2,1,0}}, allow_spmd_sharding_propagation_to_parameters={true,true,true,true,true,true}, allow_spmd_sharding_propagation_to_output={true}
```

Comment:
Note: in general it would be nice to trim HLOs as much as possible to simplify reading the tests later. Could you please do so in your newly added tests?

For instance, the part after the `HloModule <name>` on this line is not at all required for the test, so we could delete it.

In general, I'd suggest treating HLO test code as if it were real code (though SSA) and writing it such that it's easy to read. What I've found to be helpful is to remove `%` when they appear (they don't here!), trailing `.<integer>` at the end of variable names and rename those to give them meaningful names whenever possible, and re-ordering instructions for logical reasoning. Also, don't hesitate to add `// comments` when you need to explain logic in place, as those are supported by HLO.

Of course, we can be pragmatic about it, but the idea is that it should be easy for test readers to
1. easily understand why the test is structured as it is;
2. easily modify the test if they need to.